### PR TITLE
fix(curriculum): allow spaces before brackets

### LINKED ISSUE #45487: Too unforgiving tests in JS: Use Bracket Notation*
The test regexes don't allow for spaces before the brackets, which is unnecessarily strict.

## Affected page
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string
https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string

The other two bracket notation exercises are fine, since the assertions don't test the whole line.

## Your code

```
const thirdLetterOfLastName = lastName [2]; 

```

## Expected behavior
Should work just as well as the expected solution of:
```
const thirdLetterOfLastName = lastName[2]; 

```

### PR DESCRIPTION
**Pull request by raditotev**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #45487

Update regex used in tests, so that tests are more forgiving and tolerant of spaces.

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ assert(firstLetterOfLastName === 'L');
 You should use bracket notation.
 
 ```js
-assert(code.match(/firstLetterOfLastName\s*?=\s*?lastName\[.*?\]/));
+assert(code.match(/firstLetterOfLastName\s*=\s*lastName\s*\[\d{1}\]/));
```
Comment (lynxlynxlynx) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-first-character-in-a-string.md:44]: You're introducing new spacing issues with the extra changes:
```suggestion
assert(code.match(/firstLetterOfLastName\s*=\s*lastName\s*\[\s*\d\s*\]/));
```

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.md [hunk 2]
```diff
@@ -39,7 +39,7 @@ assert(thirdLetterOfLastName === 'v');
 You should use bracket notation.
 
 ```js
-assert(code.match(/thirdLetterOfLastName\s*?=\s*?lastName\[.*?\]/));
+assert(code.match(/thirdLetterOfLastName\s*=\s*lastName\s*\[\d{1}\]/));
```
Comment (lynxlynxlynx) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/use-bracket-notation-to-find-the-nth-character-in-a-string.md:42]: same here

**Comment by raditotev** [comment]
Thanks @lynxlynxlynx. I've added the suggested changes.

**Review by lynxlynxlynx — APPROVED** [review]
LGTM now, thanks!

**Review by gikf — APPROVED** [review]
Looks good :+1:

**Pull request merged**